modules/application/src/main/java/edu/northeastern/truthtree/adapter/basicInfo/BasicInfoDBAdapter.java

```diff
@@ -218,8 +285,8 @@ public JSONArray getBasicCitiesInfo() {
    * @return JSONArray representing the data on COUNTIES_URL
    */
   @Override
-  public JSONArray getBasicCountiesInfo() {
-    return JSONUtil.readJSONFile(COUNTIES_FILE_PATH);
+  public PageDTO<LocationDTO> getBasicCountiesInfo() {
+    return jsonArrayToLocationPage(JSONUtil.readJSONFile(COUNTIES_FILE_PATH));
```

Comment:
You can directly used one-time initialized counties values from static block. It will avoid reading from file in every request.

Reply:
addressed